internal/core/axis.cpp

```diff
@@ -33,7 +33,7 @@ int Axis::dimension() const noexcept(true) {
     return this->m_dimension;
 }
 
-float Axis::stride() const noexcept (true) {
+float Axis::stepsize() const noexcept (true) {
```

Comment:
I was under the impression that both changes should be in the same commit. It was quite cumbersome to ensure that all instances of stepsize and stride was working correctly. Thus, it will be quite a bit of work to split up in two separate commits. The result of the two commits will be the same as the current, but the intermediate commit would have to be tested and and checked to ensure that it is a working commit. 

I do agree with the principle here but given the cost benefits I would prefer to keep this as a single commit.

Reply:
This PR is your introduction into the way we work, so it is fine to use some time for this task, get a but frustrated and learn not to do it in the future!
It is not that expensive anyway - splitting changes is like 5 minutes, running tests locally additional 2-3 because compilation is slow. We do stuff like that all the time if we didn't think first about doing something from the start.

If you think this split would be slower for some reason it might be that something in your process is not optimal. I can help!

Reply:
Ok, I will start over with a new PR. It is significantly faster to start from master then to revert back to the stride on case.